File: test/golden/plot_golden.py
"""Plot rabe.nc output from the golden record run."""

import sys
import xarray as xr
import matplotlib.pyplot as plt

path = sys.argv[1] if len(sys.argv) > 1 else "rabe.nc"
ds = xr.open_dataset(path)
s = ds["s_tor"].values

has_sc = "lambda_SC_bB" in ds

if has_sc:
    fig, axes = plt.subplots(2, 2, figsize=(10, 8))
    ax1, ax2, ax3, ax4 = axes.flat
else:
    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 4))

ax1.plot(s, ds["Lambda_A"].values, "o-", label=r"$\Lambda_A$ (1/$\sqrt{\nu_*}$)")
ax1.plot(s, ds["Lambda_B"].values, "s-", label=r"$\Lambda_B$ (1/$\nu_*$)")
ax1.set_xlabel(r"$s_\mathrm{tor}$")
ax1.set_ylabel("coefficient [1]")
ax1.set_title("off-set")
ax1.legend()
ax1.grid(True)

ax2.plot(s, ds["Lambda_S"].values, "^-")
ax2.set_xlabel(r"$s_\mathrm{tor}$")
ax2.set_ylabel("coefficient [1]")
ax2.set_title(r"$\Lambda_\mathrm{S}$")
ax2.grid(True)

if has_sc:
    ax3.plot(s, ds["lambda_SC_bB"].values, "o-")
    ax3.set_xlabel(r"$s_\mathrm{tor}$")
    ax3.set_ylabel("coefficient [1]")
    ax3.set_title(r"$\lambda^\mathrm{SC}_{bB}$ (omnigenous Shaing-Callen)")
    ax3.grid(True)

    ax4.plot(s, ds["remainder"].values, "s-")
    ax4.set_xlabel(r"$s_\mathrm{tor}$")
    ax4.set_ylabel("coefficient [1]")
    ax4.set_title("remainder (non-omnigenous Shaing-Callen)")
    ax4.grid(True)

fig.suptitle(f"rabe output — {path}")
fig.tight_layout()
plt.savefig("rabe_output.png", dpi=150)
print("saved rabe_output.png")
plt.show()

File: test/golden/plot_golden.m
% Plot rabe.nc output from the golden record run.
% Usage: plot_golden        (reads rabe.nc in current directory)
%        plot_golden myfile.nc

pkg load netcdf;

if nargin >= 1
    path = argv(){1};
else
    path = 'rabe.nc';
end

s             = ncread(path, 's_tor');
Lambda_A     = ncread(path, 'Lambda_A');
Lambda_B     = ncread(path, 'Lambda_B');
Lambda_S = ncread(path, 'Lambda_S');

info = ncinfo(path);
varnames = {info.Variables.Name};
has_sc = any(strcmp(varnames, 'lambda_LC_bB'));

if has_sc
    lambda_LC = ncread(path, 'lambda_LC_bB');
    remainder = ncread(path, 'remainder');
    figure('Position', [100 100 900 700]);
else
    figure('Position', [100 100 900 350]);
end

if has_sc
    subplot(2, 2, 1);
else
    subplot(1, 2, 1);
end
plot(s, Lambda_A, 'o-', s, Lambda_B, 's-');
xlabel('s_{tor}');
ylabel('coefficient [1]');
title('off-set');
legend('\Lambda_A (1/\surd\nu_*)', '\Lambda_B (1/\nu_*)');
grid on;

if has_sc
    subplot(2, 2, 2);
else
    subplot(1, 2, 2);
end
plot(s, Lambda_S, '^-');
xlabel('s_{tor}');
ylabel('coefficient [1]');
title('\Lambda_{S}');
grid on;

if has_sc
    subplot(2, 2, 3);
    plot(s, lambda_LC, 'o-');
    xlabel('s_{tor}');
    ylabel('coefficient [1]');
    title('\lambda^{LC}_{bB} (omnigenous Shaing-Callen)');
    grid on;

    subplot(2, 2, 4);
    plot(s, remainder, 's-');
    xlabel('s_{tor}');
    ylabel('coefficient [1]');
    title('remainder (non-omnigenous Shaing-Callen)');
    grid on;
end

set(gcf, 'Name', ['rabe output -- ' path]);
print('rabe_output.png', '-dpng', '-r150');
printf('saved rabe_output.png\n');
waitfor(gcf);
